requirements.txt

```diff
@@ -1,8 +1,11 @@
 numpy
-Cython
+Cython<3.0.4
 pandas>=1.1.5
 xarray
 netcdf4
 pyyaml
 toolz
 joblib
+deprecated
+pyarrow<12.0
```

Comment:
I think the pyarrow version is an issue specific to running t-route with ngen, https://github.com/NOAA-OWP/t-route/issues/614 
I'm not sure if it would be best to address the issue closest to the root (here) to stop other things that use it all needing the same patch. Or if it should be addressed in ngen as that's where it appears to break things?

Reply:
@JoshCu @TrupeshKumarPatel  I tested these changes and they work fine on my end. Is the above conversation about locking in pyarrow version ongoing or is this PR ready for approval/merging?